feat: refactor Knowledge Base

eeee0717 opened the pull request:
1. migrate framework from embedjs to langchain
2. add embedding video and image
3. support multimodal embedding models(jina-clip-v2, jina-embeddings-v4)
4. support multimodal reranker model(jina-reranker-m0)

eeee0717 committed:
feat: migrate to langchain

eeee0717 committed:
feat: add website loader and sitemap loader

eeee0717 committed:
feat: add note loader

eeee0717 committed:
feat: improve

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
fix: logger error

eeee0717 committed:
feat: migrate to langchain

eeee0717 committed:
feat: add youtube loader

eeee0717 committed:
feat: add voyage embeddings

eeee0717 committed:
feat: add hybrid retriever and splitter

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
fix: merge error

eeee0717 committed:
fix: use CheerioWebBaseLoader

eeee0717 committed:
feat: add youtube loader with timestamp

eeee0717 committed:
feat: format document type

eeee0717 committed:
feat: add video block

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
fix: youtube cannot jump exact time

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
fix: i18n and merge error

eeee0717 committed:
feat: add video

eeee0717 committed:
feat: add video loader

eeee0717 committed:
feat: add local video player

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
refactor: knowledge service

eeee0717 committed:
feat: add migration to langchain

eeee0717 committed:
fix: filter video tab

eeee0717 committed:
refactor: extract processKnowledgeReferences

eeee0717 committed:
fix: structured srt splitter

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
fix: check error

eeee0717 committed:
fix

eeee0717 committed:
fix: check error

eeee0717 committed:
fix: i18n

eeee0717 commented on src/main/services/knowledge/LangChainFramework.ts:
```diff
@@ -0,0 +1,468 @@
+    }
+  }
+
+  private async createDatabase(base: KnowledgeBaseParams): Promise<void> {
```
@EurFelux 相关代码在这里

eeee0717 commented:
@EurFelux 迁移LangChain弄好了，但是有个问题，如果不自动获取嵌入维度会导致创建数据库没有size而失败，不知道之前embedjs的时候是怎么操作的，方便的话帮忙看一下。
设置如下:
![CleanShot 2025-08-10 at 16 29 14@2x](https://github.com/user-attachments/assets/9a6554e5-37d4-4202-909f-e5934269fa60)

EurFelux commented:
我记得是从api的响应获取的默认维度

eeee0717 committed:
fix: test

eeee0717 commented:
> 我记得是从api的响应获取的默认维度

我看代码里点击ok后没有获取默认维度相关的代码
![CleanShot 2025-08-10 at 16 42 10@2x](https://github.com/user-attachments/assets/33e58e74-c37b-4b9f-9a72-714b68498b60)

EurFelux committed:
refactor(store): 为旧知识库添加userDims字段确保兼容性

新知识库要求必须传入dimensions参数，而使用embedjs的旧知识库不依赖该字段。添加userDims字段来区分处理，保证旧知识库的使用安全。

EurFelux committed:
feat(知识库): 添加自动设置嵌入维度功能

- 新增 userDims 字段控制是否使用自定义维度
- 修改维度输入组件，移除自动获取按钮
- 在创建知识库时自动获取默认维度
- 更新多语言翻译文本

EurFelux committed:
Merge remote-tracking branch 'chentao/feat/langchain' into feat/langchain

EurFelux committed:
fix(store): 修复知识库迁移时userDims字段设置逻辑

当dimensions字段存在时，正确设置userDims为true

EurFelux committed:
fix(知识库框架): 根据userDims条件设置embeddings的dimensions参数

当userDims为false时，将embeddings的dimensions参数设为undefined，避免不必要的维度设置

EurFelux committed:
refactor(知识库): 将loading控制转移到onOk

EurFelux committed:
refactor(knowledge): 简化获取嵌入维度的参数传递

EurFelux committed:
refactor(知识库): 重命名fetchDimension为fetchDimensions以更准确描述功能

EurFelux committed:
feat(知识库): 添加获取嵌入维度的API服务函数

将获取嵌入维度的逻辑从组件移至ApiService中，便于复用

EurFelux committed:
feat(useKnowledge): 添加获取知识库维度的错误处理逻辑

在迁移知识库时，如果未定义维度则尝试从API获取，失败时记录错误并显示提示信息

EurFelux committed:
test: 在测试数据中添加dimensions字段

EurFelux committed:
feat(websearch): 添加嵌入维度自动设置开关并设为默认

为压缩配置添加用户自定义嵌入维度开关，默认使用自动设置

EurFelux committed:
fix(WebSearchService): 修复知识库迁移时未检查dimensions的问题

添加对旧知识库的迁移支持，当dimensions不存在时自动获取并更新配置

EurFelux committed:
feat(记忆设置): 添加自动获取嵌入模型维度功能

添加自动获取嵌入模型维度的开关选项，当开启时自动从API获取维度值，关闭时手动输入维度值

EurFelux committed:
feat(内存设置): 添加更新设置失败的错误提示并优化加载状态处理

在内存设置模态框中添加更新设置失败的错误提示，使用新的错误格式化工具。同时优化加载状态的显示逻辑，确保加载状态在请求完成后被正确清除。

EurFelux committed:
fix(内存服务): 根据配置自动设置嵌入维度或使用默认值

EurFelux committed:
fix(store): 为websearch和memory添加embeddingDimensions默认值

迁移旧配置时添加embeddingDimensions默认值1024，并处理memory的isAutoDimensions逻辑

EurFelux committed:
docs(i18n): 更新多语言翻译文件中的嵌入维度相关文本

完善中文繁体、日语、英语、俄语、葡萄牙语、希腊语、西班牙语和法语翻译文件
添加缺失的翻译条目并修正现有翻译

EurFelux committed:
fix(i18n): 修正日语和俄语翻译中的错误信息

将日语和俄语翻译中的"更新设置失败"错误信息分别修正为正确的本地化翻译

EurFelux committed:
test(InputEmbeddingDimension): 更新测试用例和快照以匹配默认维度值

refactor(GeneralSettingsPanel): 重构antd mock实现并添加维度自动设置开关

更新测试用例以使用默认维度值1024作为占位符，并移除不再需要的刷新按钮相关测试
重构antd mock实现使用动态导入，同时在GeneralSettingsPanel中添加维度自动设置开关功能

eeee0717 committed:
chore: add auto vacuum

EurFelux commented:
> > @alephpiece 适应langchain需要dimensions字段，大改过了
> 
> langchain 的这个参数不是可选的吗？

构建向量数据库需要确定的维度

eeee0717 committed:
feat: migrate to faiss

eeee0717 committed:
fix: save to db

eeee0717 committed:
fix: trash bug

eeee0717 committed:
fix: directory task

EurFelux commented:
> > > > @alephpiece 适应langchain需要dimensions字段，大改过了
> > > 
> > > 
> > > langchain 的这个参数不是可选的吗？
> > 
> > 
> > 构建向量数据库需要确定的维度
> 
> 那是 libsql 自己需要的，langchain 只是适配它而已吧？从 langchain 这一层开始，根本不关心底下存储的表里的维度值是多少吧？我感觉也不用要求在使用知识库时传入 dimension 呀，最好的保持维度一致性方法不是运行时检测吗？就算没有直接获取输出维度的方式，无论如何第一次发送嵌入请求也能知道向量维度了。这样一来，用户不必关心底下的细节，只要关心想不想在请求体里面加入 dimension，因为无论他加不加，最后输出维度都是自动处理好的。

langchain需要手动管理向量数据库，之前这部分工作是embedjs完成的。

我还是坚持把dimensions字段做成必选属性，另外用字段控制参数传递。数据在需要用的时候能直接用是最好的。为了实现的时候省事就把不同的状态耦合在一起，需求变更的时候会多出很多不必要的工作量。

而且对于用户来说，这个方案也没有增加心智负担，如果不需要自定义嵌入维度，仍然一切都是底层自动完成的。

alephpiece commented:
> > > > > @alephpiece 适应langchain需要dimensions字段，大改过了
> > > > 
> > > > 
> > > > langchain 的这个参数不是可选的吗？
> > > 
> > > 
> > > 构建向量数据库需要确定的维度
> > 
> > 
> > 那是 libsql 自己需要的，langchain 只是适配它而已吧？从 langchain 这一层开始，根本不关心底下存储的表里的维度值是多少吧？我感觉也不用要求在使用知识库时传入 dimension 呀，最好的保持维度一致性方法不是运行时检测吗？就算没有直接获取输出维度的方式，无论如何第一次发送嵌入请求也能知道向量维度了。这样一来，用户不必关心底下的细节，只要关心想不想在请求体里面加入 dimension，因为无论他加不加，最后输出维度都是自动处理好的。
> 
> langchain需要手动管理向量数据库，之前这部分工作是embedjs完成的。
> 
> 我还是坚持把dimensions字段做成必选属性，另外用字段控制参数传递。数据在需要用的时候能直接用是最好的。为了实现的时候省事就把不同的状态耦合在一起，需求变更的时候会多出很多不必要的工作量。
> 
> 而且对于用户来说，这个方案也没有增加心智负担，如果不需要自定义嵌入维度，仍然一切都是底层自动完成的。

可能我理解的有问题😂我理解的是 langchain 的 vectorstore 和 libsql 数据库完全是两回事，不用 libsql 是不需要 dimension 的，向量数据库本身只是个抽象层而已。创建 libsql vectorstore 需要指明的也是列名，这个 dimension 完全是 libsql 自己为了支持向量存储新增的东西。

EurFelux commented:
@eeee0717 faiss怎么样，其他数据库不用dimensions参数能建立吗？

eeee0717 commented:
faiss可以不用dimension，因为直接存的json格式

EurFelux commented:
> faiss可以不用dimension，因为直接存的json格式

faiss效果怎么样？决定用faiss了可以把ui改动revert掉。

EurFelux committed:
Revert "test(InputEmbeddingDimension): 更新测试用例和快照以匹配默认维度值"

This reverts commit 7a738d0aa35176ce5d16091e4971640e28eb0812.

EurFelux committed:
Revert "fix(store): 为websearch和memory添加embeddingDimensions默认值"

This reverts commit d6aa10b6dfa87940283f543aa9d423012a0802e3.

EurFelux committed:
Revert "fix(内存服务): 根据配置自动设置嵌入维度或使用默认值"

This reverts commit c2ecac4871c9537a0838d6bf05e0fc937d7b1348.

EurFelux committed:
Revert "feat(内存设置): 添加更新设置失败的错误提示并优化加载状态处理"

This reverts commit 1a7be153a5ddc722098d3dfceffedf19555b9634.

EurFelux committed:
Revert "feat(记忆设置): 添加自动获取嵌入模型维度功能"

This reverts commit 357822915dac5d890f0105003f5f015dc0c9ccbb.

EurFelux committed:
Revert "fix(WebSearchService): 修复知识库迁移时未检查dimensions的问题"

This reverts commit 13cbcec4a9831c89c5e0ef3c72b95b667d75e521.

EurFelux committed:
Revert "feat(websearch): 添加嵌入维度自动设置开关并设为默认"

This reverts commit 8ea0ccc3990242bcfc72ad69509d4feeeb0414ba.

EurFelux committed:
Revert "test: 在测试数据中添加dimensions字段"

This reverts commit a2dde3de38fc4f6c8f1ba8143d6ee31cc66f1f01.

EurFelux committed:
Revert "feat(useKnowledge): 添加获取知识库维度的错误处理逻辑"

This reverts commit 24f83f907a66c42ed3a34cb27c41a2ac11709042.

EurFelux committed:
Revert "feat(知识库): 添加获取嵌入维度的API服务函数"

This reverts commit 1a5a3257825dce7f8fcebc275f062bbb0fa457a1.

EurFelux committed:
Revert "refactor(知识库): 重命名fetchDimension为fetchDimensions以更准确描述功能"

This reverts commit 08bf19559df391f65a88a1479070ea1524fb26b1.

EurFelux committed:
Revert "refactor(knowledge): 简化获取嵌入维度的参数传递"

This reverts commit 35fef47808fb3ec97d5d88ade292caf974bde49b.

EurFelux committed:
Revert "refactor(知识库): 将loading控制转移到onOk"

This reverts commit 56080a997605192220d3319859815559c26a2599.

EurFelux committed:
Revert "fix(知识库框架): 根据userDims条件设置embeddings的dimensions参数"

This reverts commit 45456a4595d33c99952c05383325e27c9e7d77b5.

EurFelux committed:
Revert "fix(store): 修复知识库迁移时userDims字段设置逻辑"

This reverts commit 4ceb2be37ddf1cbf11357041c5205e3db1b1f48b.

EurFelux committed:
Revert "fix(i18n): 修正日语和俄语翻译中的错误信息"

This reverts commit c9f03b92c2292e35a9344b03cb85c81b6b957f92.

EurFelux committed:
Revert "docs(i18n): 更新多语言翻译文件中的嵌入维度相关文本"

This reverts commit 9862001dd7b2b0891f2a112f7a937ead3ee1fbcc.

EurFelux committed:
Revert "feat(知识库): 添加自动设置嵌入维度功能"

This reverts commit e3167aa7ae9338a556e4f03827cd360d80498931.

EurFelux committed:
Revert "refactor(store): 为旧知识库添加userDims字段确保兼容性"

This reverts commit 6250053f70aa835b8df46c74b1103d63e0f7f846.

eeee0717 committed:
fix: remove directory loader

eeee0717 committed:
fix: score

eeee0717 committed:
fix: setting popup style

eeee0717 committed:
feat: add retriver weight

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
fix

eeee0717 committed:
chore

eeee0717 committed:
chore(version): 1.5.6-beta.1

eeee0717 commented on package.json:
```diff
@@ -73,13 +73,21 @@
     "@libsql/client": "0.14.0",
     "@libsql/win32-x64-msvc": "^0.4.7",
     "@strongtz/win32-arm64-msvc": "^0.4.7",
+    "cheerio": "^1.1.2",
```
@beyondkmp 付哥可以帮忙看看这里哪些可以放在devDependencies中吗，不太知道哪些是可放哪些不可放😂

eeee0717 committed:
fix: migration error

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
feat: add image loader

eeee0717 committed:
fix: ollama

eeee0717 committed:
feat: add multimodal embeddings

eeee0717 committed:
feat: add image callback

eeee0717 committed:
fix

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
feat: finish knowledge image

eeee0717 committed:
fix: image item component

eeee0717 committed:
fix: i18n

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
fix: remove pdfjs-dist

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
feat: add multimodal reranker

eeee0717 committed:
fix

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
fix: duplicate video

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
fix: i18n

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
feat: support markdown loader

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
Merge branch 'main' into feat/langchain

alephpiece commented:
这个迁移到新架构的按钮，提示文本准确吗？这个文本好像还是知识库设置变化引起的迁移使用的文本

<img width="420" height="177" alt="image" src="https://github.com/user-attachments/assets/66657b94-b615-44ae-a242-117db0459c12" />

迁移完成之后，是不是可以隐藏按钮了？

<img width="242" height="47" alt="image" src="https://github.com/user-attachments/assets/a49a3dbf-aeec-4962-9a44-1f7b5ce3374d" />

eeee0717 commented:
@alephpiece 
1. 文本描述准确的，和设置改变后的迁移逻辑一致，会生成一个副本，原先的知识库可以由用户自行决定是否删除
2. 我觉得可以不用隐藏，可以提醒用户这是原知识库，如果要可隐藏还得多加一个字段

eeee0717 committed:
Merge branch 'main' into feat/langchain

alephpiece commented:
旧知识库修改配置之后，是不是也可以直接在保存配置的时候迁移到新知识库？

eeee0717 committed:
fix: search popup show

eeee0717 commented:
> > @alephpiece
> > 
> > 1. 文本描述准确的，和设置改变后的迁移逻辑一致，会生成一个副本，原先的知识库可以由用户自行决定是否删除
> > 2. 我觉得可以不用隐藏，可以提醒用户这是原知识库，如果要可隐藏还得多加一个字段
> 
> 是我理解错了，这是迁移一个是吗？我还以为是迁移所有的，有按钮用来迁移所有的吗？

目前没有迁移所有的按钮

eeee0717 commented:
> 小屏幕上有点拥挤
> 
> <img alt="image" width="807" height="174" src="https://private-user-images.githubusercontent.com/22237751/483011633-af39e6a2-bb76-44d3-838c-46241470fe9f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYzNjU4MzUsIm5iZiI6MTc1NjM2NTUzNSwicGF0aCI6Ii8yMjIzNzc1MS80ODMwMTE2MzMtYWYzOWU2YTItYmI3Ni00NGQzLTgzOGMtNDYyNDE0NzBmZTlmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI4VDA3MTg1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdkM2E0MTA4MzMwODNjNGQxOWM4MTM2YjY5MzNhN2JkYTFmMjYwZTU1OGYyMmI0NTgwMmRlOTFlYWRkYmYwYzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xYqzCx8TcojQwfboMB0qf4GI8sLGwRs1PRO9qXx2Qjg">

![CleanShot 2025-08-28 at 15 24 44@2x](https://github.com/user-attachments/assets/9580aba5-2cd5-4cd8-8310-f101ff1f71d0)
已修改

eeee0717 committed:
fix： ResponsiveButton

eeee0717 commented:
> > > @alephpiece
> > > 
> > > 1. 文本描述准确的，和设置改变后的迁移逻辑一致，会生成一个副本，原先的知识库可以由用户自行决定是否删除
> > > 2. 我觉得可以不用隐藏，可以提醒用户这是原知识库，如果要可隐藏还得多加一个字段
> > 
> > 
> > 是我理解错了，这是迁移一个是吗？我还以为是迁移所有的，有按钮用来迁移所有的吗？
> 
> 目前没有迁移所有的按钮，需要加一个吗

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
fix: prettier

EurFelux committed:
refactor(embeddings): 使用zod验证Jina模型并提取类型检查函数

将JinaEmbeddings的模型类型定义从接口迁移到zod schema
添加isJinaEmbeddingsModel类型检查函数替代字符串包含检查
简化相关接口的类型定义

EurFelux committed:
refactor(embeddings): 优化OllamaEmbeddings的初始化逻辑

简化OllamaEmbeddings的创建代码，提取公共配置
移除重复的requestOptions设置
支持可选的apiKey授权头

EurFelux committed:
refactor(embeddings): 使用SystemProviderIds替换硬编码的provider字符串

EurFelux committed:
fix(langchain/embeddings): 修复MultiModalEmbeddings中sdk类型检查问题

确保sdk实例为JinaEmbeddings类型，否则抛出错误

EurFelux committed:
feat(文件处理): 添加图片MIME类型验证并改进文件加载

为图片处理添加MIME类型验证schema，确保只处理支持的图片类型
将MarkdownLoader的文件读取方法替换为自动编码版本
完善base64Image返回类型并添加图片类型检查逻辑

EurFelux committed:
fix(gemini): 移除不必要的类型断言

EurFelux committed:
fix(openai): 移除image_url中不必要的类型断言

EurFelux committed:
build: 添加 mime-types 及其类型定义依赖

EurFelux committed:
feat(文件工具): 添加mime类型到文件类型的转换功能

EurFelux committed:
fix(VideoPopup): 修复文件上传类型判断并添加错误处理

使用 mime2type 替代直接类型转换，增加文件类型验证
当上传非 File 类型时记录错误日志

EurFelux committed:
refactor(types): 将 RemoteFile 接口改为联合类型以明确区分不同文件类型

EurFelux committed:
fix(gemini): 修复文件上传验证和类型安全转换问题

添加对上传文件的验证逻辑，确保文件类型和必要字段存在
将类型断言改为类型安全的 satisfies 操作符

EurFelux committed:
refactor(anthropic): 移除共享mime模块并内联图片类型验证

将图片MIME类型验证逻辑从共享模块移至Anthropic客户端内部
简化依赖关系，避免不必要的模块引用

EurFelux committed:
refactor(类型安全): 改进getStoreSetting的类型定义并移除不必要的类型断言

优化getStoreSetting函数，使用泛型确保返回类型与key匹配
移除KnowledgeQueue中不必要的类型断言，提高代码可读性

EurFelux committed:
feat(知识库): 扩展知识条目类型并优化类型检查

添加文件、视频和笔记类型的知识条目类型定义
实现类型守卫函数用于类型检查
更新数据库和钩子逻辑以使用新类型

EurFelux commented on src/renderer/src/pages/home/Messages/Blocks/index.tsx:
```diff
@@ -60,15 +67,31 @@ interface Props {
       }
+    } else if (currentBlock.type === MessageBlockType.VIDEO) {
+      // 对于VIDEO类型，按相同filePath分组
+      const videoBlock = currentBlock as VideoMessageBlock
```
还有很多as，都需要修改一下，严格做类型检查

eeee0717 committed:
fix: voyage ai batch size

eeee0717 committed:
fix: type check

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
fix: lint error

eeee0717 committed:
fix: typecheck error

EurFelux commented on src/renderer/src/components/Popups/VideoPopup.tsx:
```diff
@@ -0,0 +1,205 @@
+          uploadType="video"
+          accept="video/*"
+          title="视频文件:"
+          hint="支持 MP4, AVI, MOV 等视频格式"
```
需要i18n

EurFelux commented on src/renderer/src/pages/files/FileItem.tsx:
```diff
@@ -72,6 +73,10 @@ const getFileIcon = (type?: string) => {
     return <FolderOpenFilled />
   }
 
+  if (['.mp4'].includes(ext)) {
```
现在只支持mp4吗？这个列表可以做成常量维护

EurFelux commented on src/renderer/src/pages/home/Messages/MessageVideo.tsx:
```diff
@@ -0,0 +1,110 @@
+
+      default:
+        if (block.filePath) {
+          logger.warn(`未知的视频类型: ${block.metadata?.type}, 但因存在 filePath 将尝试渲染为本地视频。`)
```
日志统一用英文

eeee0717 committed:
fix

eeee0717 committed:
fix: knowledge as

eeee0717 committed:
fix: knowledge as

eeee0717 committed:
revert: multimodal embeddings

eeee0717 committed:
fix: i18n

eeee0717 committed:
Merge branch 'main' into feat/langchain

EurFelux committed:
feat(类型工具): 添加类型安全的objectValues函数并优化reranker类型检查

使用objectValues替换Object.values以提供更好的类型推断
优化isTEIProvider和isKnownProvider函数的参数类型

EurFelux committed:
feat(知识模块): 为笔记类型添加可选的sourceUrl字段

去除LangChainFramework中sourceUrl的类型断言，直接更改类型定义

EurFelux committed:
fix(hooks): 修复定时器清理问题并优化知识库检查逻辑

修复useTimer中清理定时器的函数调用方式
使用useTimer统一管理知识库检查的定时任务，避免内存泄漏

eeee0717 committed:
fix(useKnowledge): use checkAllBases

eeee0717 committed:
Merge branch 'main' into feat/langchain

eeee0717 committed:
Merge branch 'main' into feat/langchain

Commit:
Merge branch 'main' into feat/langchain

eeee0717 committed:
feat: support process knowledge references

eeee0717 committed:
chore: fix lint

eeee0717 committed:
fix: typecheck

eeee0717 committed:
Merge branch 'main' into feat/langchain

kawainotu commented:
> [!NOTE]
> This issue/comment/review was translated by Claude.

I'd like to ask, after this architectural adjustment, does calling the knowledge base seem to consume more model usage? I'm using the Volcano Engine model.

---
<details>
<summary>Original Content</summary>
想问一下，这次架构调整之后，调用知识库感觉模型用量偏大了?用的是火山引擎的模型。
</details>

eeee0717 commented:
> [!NOTE]
> This issue/comment/review was translated by Claude.

This PR has been reverted, currently still using the original architecture.

---
<details>
<summary>Original Content</summary>
这个pr已经revert了，目前还是使用原先的架构
</details>
